lib/csv.rb

```diff
@@ -1058,9 +1058,11 @@ def filter(input=nil, output=nil, **options)
         end
       end
       # build input and output wrappers
-      input  = new(input  || ARGF,    **in_options)
+      input  = parse(input  || ARGF, **in_options)
```

Comment:
We should not use `parse` here because `parse` parses all input at once. It may consume large resource.